packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -196,19 +209,29 @@ static void InterpretInfoWindow(id<FLTGoogleMapMarkerOptionsSink> sink, NSDictio
       @throw exception;
     }
   } else if ([iconData[0] isEqualToString:@"fromBytes"]) {
-    if (iconData.count == 2) {
+    if ((iconData.count == 2) || (iconData.count == 3)) {
+      key = nil;
+      if (iconData.count == 3) {
+          key = [NSString stringWithFormat:@"%@/%f", iconData[0], iconData[2]];
```

Comment:
I think it is not good idea to make `fromBytes`:
- This cache can be both for Android and iOS.
- It is more likely that the user should cache the BitmapDescriptor, if necessary.
- If you add something by key, you should be able to update / reset the cache

Reply:
+1 to that, I think we need to have an identifier per bitmap and make sure we only transmit the bytes over the method channel once.

In a scenario with a large number of `fromBytes` markers, in addition to the high memory usage here we also have much more bytes going over the method channel than needed.

Reply:
Sorry I spoke too early :) looks like the tag is that ID I'm asking for, though we're still sending the same bytes over the method channel multiple times. What do you think about making it so that each BitmapDescriptor has a unique ID? then I think we can make it so that we're only transmitting the bytes over the method channel once per bitmap?

It will require some more work, but it will have gains for all existing users (on all platforms) even without having them add a tag per marker. If we envision this as the better solution we'll want to eventually have I'd be very hesitant to introduce the tag parameter right now which will be a breaking change to remove later.